### ISSUE TITLE
Ability to set logging level in unit tests for Gradle plugin

### ISSUE THREAD

**Issue by szpak** [gradle/gradle#2030]
It seems to be unable to set custom logging level (by default LIFECYCLE+) for unit test of the Gradle plugin (which internally use SLF4J Logger). It makes it hard to diagnose issues with the plugin logic running those tests.

### Expected Behavior
I would like to be able to set INFO (or DEBUG) for **unit** tests (without Gradle being involved at all) via a logger configuration or an API call before tests execution.

### Current Behavior
Only LIFECYCLE+ log statements are logged which very often is not enough.

### Context
In my plugin in addition to functional tests I have unit tests of the plugin logic. I would like to be able to change logging level to see also INFO messages (by default only LIFECYCLE+ are visible).

In my code I use standard Slf4j's Logger. First I tried with `logback.xml`, but Gradle seems to use its own slf4j implementation (`OutputEventListenerBackedLogger`) which wins in unit tests in runtime (although the code is Gradle agnostic) and I need (in a very ugly way) to get context from the logger and set in it (and in addition in `OutputEventRenderer`) a desired level.

Having no answer on the [forum](https://discuss.gradle.org/t/setting-logging-level-in-unit-tests-for-gradle-plugin/22455) I assume it is currently not possible to achieve.

### Your Environment
Gradle 3.5 (it is reproducible with any modern Gradle version).

**Comment by bmuschko** [comment]
Answered the forum post. I am not sure if there's a good solution.

**Comment by szpak** [comment]
Yup, therefore please treat is not as a question, but as a feature request. It should be possible to achieve that without working on private fields in Gradle classes.

**Comment by szpak** [comment]
It would still be valuable for people writing lower level tests (without TestKit) for business logic used in the Gradle plugin to be able to set the logging level.

**Comment by kanchanasrishti** [comment]
Hi, is there any update on this issue?

I am most probably facing the same issue.

We build and run the tests in our java application using gradle.
We program against the slf4j-interface in our application code and then plug in log4j2 framework as an implementation of that interface, which handles the actual logging.

The runtime logging on the target hardware works fine. But, JUnit tests are logged at WARN level only when running gradle test command. Changing root level logging in log4j-test.xml has no effect on the logs output from the JUnit tests.

It looks like gradle chooses to use it's own logging implementation.

Any help in this regard is highly appreciated.

Thank you in advance.

**Comment by felipebn** [comment]
Would be great to be able to do it on a straightforard way, we found a workaround to change the level by running the function below before tests:

```kotlin
	import org.gradle.api.logging.LogLevel
	import org.gradle.internal.logging.sink.OutputEventRenderer
	import org.gradle.internal.logging.slf4j.OutputEventListenerBackedLoggerContext
	import org.slf4j.LoggerFactory
	//...
    private fun setupGradleLogger() =
        with((LoggerFactory.getILoggerFactory() as OutputEventListenerBackedLoggerContext)) {
            level = LogLevel.DEBUG
            (outputEventListener as OutputEventRenderer).configure(LogLevel.DEBUG)
        }

```

This changes the logLevel in the existing Logger context.

**Comment by rnc** [comment]
This issue is still relevant
